lib/slither/generator.rb

```diff
@@ -8,20 +8,22 @@ def generate(data)
       @builder = []
       @definition.sections.each do |section|
         content = data[section.name]
-        if content
-          content = [content] if content.is_a?(Hash)
-          raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") unless content.any?
+        content = [content] if content.is_a?(Hash) || content.nil?
+        raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") if !content || empty_required_content(content, section)
 
+        if content.any?
           content.each do |row|
             @builder << section.format(row)
           end
-        else
-          unless section.optional
-            raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.")
-          end
         end
       end
```

Comment:
Wdyt of

```ruby
@definition.sections.each do |section|
  content = data[section.name]
  raise(Slither::RequiredSectionEmptyError, "Required section '#{section.name}' was empty.") if !content&.any? && !section.optional

  content = [content] if content.is_a?(Hash)
  content.each do |row|
    @builder << section.format(row)
  end
end
```

Reply:
The iteration on the content should be safe navigated, because we do not want to raise when content is empty, but it will raise when want to call each on nil.
Also I would use none? instead of !any?

Reply:
> The iteration on the content should be safe navigated, because we do not want to raise when content is empty, but it will raise when want to call each on nil.

Niiice 👍 

> Also I would use none? instead of !any?

Think we can't use that with the safe navigation, cuz if we're passing `nil`, then it will be `nil && true` or `nil && false` and because it's a falsey value, we won't raise.

Reply:
Added in https://github.com/Kaligo/slither/pull/3/commits/775d452592a05724cd969bf5a21586931985d40f and bumped to `master`.